internal/elfexec/elfexec.go

```diff
@@ -189,9 +188,15 @@ func GetBase(fh *elf.FileHeader, loadSegment *elf.ProgHeader, stextOffset *uint6
 	switch fh.Type {
 	case elf.ET_EXEC:
 		if loadSegment == nil {
-			// Fixed-address executable, no adjustment.
+			// Assume fixed-address executable and so no adjustment.
 			return 0, nil
 		}
+		if stextOffset == nil && start > 0 && start < 0x8000000000000000 {
+			// A regular user-mode executable. Compute the base offset using same
+			// arithmetics as in ET_DYN case below, see the explanation there.
+			return start - offset + loadSegment.Off - loadSegment.Vaddr, nil
```

Comment:
I think this formula is the general way to compute the base, at least for user mode addresses, so I am happy with this change.
But, can you add a comment on what stextOffset represents?

Reply:
Done.